Nova documentação de Notificações (feature/new-notifications) (#2426)
mercadopago/devsite-docs: guides/additional-content/your-integrations/webhooks.pt.md

# Webhooks

Webhooks (também conhecido como retorno de chamada web) são um método simples que permite a uma aplicação ou sistema fornecer informações em tempo real sempre que um evento ocorre. É uma forma passiva de receber dados entre dois sistemas por meio de uma solicitação `HTTP POST`.

As notificações Webhooks podem ser configuradas para cada uma das aplicações criadas em  [Suas integrações](/developers/panel/app). Você também poderá configurar uma URL de teste que, junto com suas credenciais de teste, permitirá testar o funcionamento correto das suas notificações antes de sair à produção.

Uma vez configuradas, as notificações Webhooks serão enviadas sempre que ocorrer um ou mais eventos cadastrados. Isso evita a necessidade de verificações constantes, prevenindo a sobrecarga do sistema e a perda de dados em situações críticas. 

Para configurar as notificações Webhooks, escolha entre uma das opções abaixo:

| Tipo de configuração | Descrição |
|---|---|

> | [Configuração via Suas integrações](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoviasuasintegraes) | Permite configurar notificações para cada aplicação, identificar contas diferentes se necessário, e validar a origem da notificação utilizando uma assinatura secreta (exceto para notificações de integrações com Código QR). |

Author: ```suggestion
| [Configuração via Suas integrações](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoviasuasintegraes) | Permite configurar notificações para cada aplicação, identificar contas diferentes se necessário, e validar a origem da notificação utilizando uma assinatura secreta----[mla, mlb, mlu, mlc]----(exceto para notificações de integrações com Código QR)------------. |
```

> | [Configuração via Suas integrações](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoviasuasintegraes) | Permite configurar notificações para diversos tópicos, identificar contas diferentes se necessário, e validar a origem da notificação utilizando uma assinatura secreta (exceto para notificações de integrações com Código QR). |

Author: ```suggestion
| [Configuração via Suas integrações](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoviasuasintegraes) | Permite configurar notificações para cada aplicação, identificar contas diferentes se necessário, e validar a origem da notificação utilizando uma assinatura secreta (exceto para notificações de integrações com Código QR). |
```

> | [Configuração durante a criação de pagamentos](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoduranteacriaodepagamentos) | Permite a configuração específica das notificações para cada pagamento. Não é permitido configurar para integrações com Mercado Pago Point e nem para integrações com Mercado Pago Delivery. |

Author: ```suggestion
| [Configuração durante a criação de pagamentos](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoduranteacriaodepagamentos) | Permite a configuração específica das notificações para cada pagamento, preferência ou pedidos comerciais. Não é permitido configurar para integrações com Mercado Pago Point e nem para integrações com Mercado Pago Delivery. |
```

> | [Configuração durante a criação de pagamentos](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoduranteacriaodepagamentos) | Permite a configuração específica das notificações para cada pagamento, preferência ou pedidos comerciais. Não é permitido configurar para integrações com Mercado Pago Point e nem para integrações com Mercado Pago Delivery. |

Author: ```suggestion
| [Configuração durante a criação de pagamentos](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoduranteacriaodepagamentos) | Permite a configuração específica das notificações para cada pagamento, preferência ou pedidos comerciais. ----[mla]----Não é permitido configurar para integrações com Mercado Pago Point e nem para integrações com Mercado Pago Delivery----------------[mlb, mlm]----Não é permitido configurar para integrações com Mercado Pago Point------------.|
```

> WARNING
>
> Importante
>
> As URLs configuradas durante a criação do pagamento terão prioridade sobre aquelas configuradas através de Suas integrações. 

Uma vez que as notificações estiverem configuradas, consulte as [ações necessárias após receber uma notificação](/developers/pt/docs/your-integrations/notifications/webhooks#aesnecessriasapsreceberumanotificao) para validar se foram devidamente recebidas.

## Configuração via Suas integrações 

Configure notificações para cada aplicação diretamente em [Suas integrações](/developers/panel/app) de forma eficiente e segura. Nesta documentação, explicaremos como:

1. Indicar URLs e configurar eventos
2. Validar origem da notificação
3. Simular o recebimento da notificação

> WARNING
>
> Importante
>

> > Este método de configuração não está disponível para integrações com Código QR e nem Assinaturas. Para configurar notificações com alguma dessas duas integrações, utilize o método [Configuração durante a criação de pagamentos](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoduranteacriaodepagamentos).

Author: ```suggestion
> Este método de configuração não está disponível para integrações com----[mla, mlb, mlc, mlu]----Código QR e nem------------Assinaturas. Para configurar notificações com alguma dessas duas integrações, utilize o método [Configuração durante a criação de pagamentos](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoduranteacriaodepagamentos).
```

Para isso, siga as etapas descritas abaixo.

1. Acesse [Suas integrações](/developers/panel/app) e selecione a aplicação para a qual deseja ativar as notificações. Caso ainda não tenha criado uma aplicação, acesse a [documentação Painel do Desenvolvedor](/developers/pt/docs/your-integrations/dashboard) e siga as instruções.
2. No menu à esquerda, vá até **Webhooks > Configurar notificações** e configure as URLs que serão usadas para receber as notificações. Recomendamos utilizar uma URL diferente para o modo de teste e o modo produção:
    * **URL modo teste:** fornece uma URL que permite testar o correto funcionamento das notificações dessa aplicação durante a fase de teste ou desenvolvimento. O teste dessas notificações deverá ser realizado exclusivamente com as **credenciais de teste de usuários produtivos**.
    * **URL modo produção:** fornece uma URL para receber notificações com sua integração produtiva. Essas notificações deverão ser configuradas com **credenciais produtivas**.

![webhooks](/images/dashboard/webhooks-pt.png)

> NOTE
>
> Nota
> 

> > Caso seja necessário identificar múltiplas contas, adicione o parâmetro `?cliente=(nomedovendedor) endpoint` ao final da URL indicada para identificar os vendedores.

Author: ```suggestion
> Caso seja necessário identificar múltiplas contas, adicione o parâmetro `?cliente=(nomedovendedor)` ao final da URL indicada para identificar os vendedores.
```

3. Selecione os  **eventos** para os quais deseja receber notificações em formato `json` através de um `HTTP POST` para a URL especificada anteriormente. Um evento pode ser qualquer atualização no objeto relatado, incluindo alterações de status ou atributos. Consulte a tabela abaixo para ver os eventos configuráveis, considerando a solução do Mercado Pago integrada e suas necessidades de negócio.

| Eventos | Nome em Suas integrações | Tópico | Produtos associados |
|---|---|---|---|

> | Criação e atualização de pagamentos | Pagamentos | `payment` | Checkout ----[mlb]----Transparente ----------------[mla, mlu, mlc, mlm, mco, mpe]----API------------<br>Checkout Pro<br>Checkout Bricks<br>Assinaturas<br>----[mla, mlm, mlb]----MP Point------------<br>Wallet Connect |

Reviewer: ```suggestion
| Criação e atualização de pagamentos | Pagamentos | `payment` | Checkout ----[mlb]----Transparente ----------------[mla, mlu, mlc, mlm, mco, mpe]----API------------<br>Checkout Pro<br>Checkout Bricks<br>Assinaturas<br>----[mla, mlm, mlb]----Mercado Pago Point------------<br>Wallet Connect |
```

| Pagamento recorrente de uma assinatura (criação - atualização) | Planos e assinaturas | `subscription_authorized_payment` | Assinaturas |
| Vinculação de uma assinatura (criação - atualização) | Planos e assinaturas | `subscription_preapproval` | Assinaturas |
| Vinculação de um plano de assinatura (criação - atualização) | Planos e assinaturas | `subscription_preapproval_plan` | Assinaturas |
| Vinculação e desvinculação de contas mp-connect | Vinculação de aplicações | `mp-connect` | Todos os produtos que tenham  OAuth implementado |
----[mla, mlm, mlb]----| Finalização, cancelamento ou erros ao processar intenções de pagamento de dispositivos Mercado Pago Point. | Integrações Point | `point_integration_wh` | Mercado Pago Point |------------
| Transações com Wallet Connect | Wallet Connect | `wallet_connect` | Wallet Connect |
| Alertas de fraude após o processamento de um pedido | Alertas de fraude | `stop_delivery_op_wh` | Checkout ----[mlb]----Transparente ----------------[mla, mlu, mlc, mlm, mco, mpe]----API------------<br>Checkout Pro |

> | Criação de estornos e reclamações | Reclamações | `topic_claims_integration_wh` | Checkout ----[mlb]----Transparente ----------------[mla, mlu, mlc, mlm, mco, mpe]----API------------<br>Checkout Pro<br>Checkout Bricks<br>Assinaturas<br>----[mla, mlm, mlb]----MP Point------------<br>Código QR<br>Wallet Connect |

Reviewer: ```suggestion
| Criação de estornos e reclamações | Reclamações | `topic_claims_integration_wh` | Checkout ----[mlb]----Transparente ----------------[mla, mlu, mlc, mlm, mco, mpe]----API------------<br>Checkout Pro<br>Checkout Bricks<br>Assinaturas<br>----[mla, mlm, mlb]----Mercado Pago Point------------<br>Código QR<br>Wallet Connect |
```

| Recuperação e atualização de informações de cartões no Mercado Pago | Card Updater | `topic_card_id_wh` | Checkout Pro<br>Checkout ----[mlb]----Transparente ----------------[mla, mlu, mlc, mlm, mco, mpe]----API------------<br>Checkout Bricks |
| Criação, fechamento ou expiração de ordens comerciais. | Ordens comerciais | `topic_merchant_order_wh` | Checkout Pro<br>Código QR  |
| Abertura de _chargebacks_, mudanças de status e modificações referentes às liberações de dinheiro | Chargebacks | `topic_chargebacks_wh` | Checkout Pro<br>Checkout ----[mlb]----Transparente ----------------[mla, mlu, mlc, mlm, mco, mpe]----API------------<br>Checkout Bricks |

> ----[mla]----| Criação, atualização ou cancelamento de pedidos. | Delivery (proximity marketplace) | `delivery` | MP Delivery |------------

Reviewer: ```suggestion
----[mla]----| Criação, atualização ou cancelamento de pedidos. | Delivery (proximity marketplace) | `delivery` | Mercado Pago Delivery |------------
```

> WARNING
>
> Importante
>
> Em caso de dúvidas sobre quais tópicos ativar ou quais eventos serão notificados, , consulte a documentação [Informações adicionais sobre notificações](/developers/pt/docs/your-integrations/notifications/additional-info). 

5. Por fim, clique e **Salvar** para gerar uma **assinatura secreta** exclusiva para a sua aplicação, permitindo validar a autenticidade das notificações recebidas e garantir que tenham sido enviadas pelo Mercado Pago. A assinatura gerada não tem prazo de validade e sua renovação periódica não é obrigatória, embora seja altamente recomendável. Para renová-la, clique no botão de **redefinição** ao lado da assinatura.

> 

Author: ```suggestion

----[mla, mlb, mlu, mlc]----

```

> WARNING
> 
> Importante
> 

> > As notificações de Código QR não podem ser verificadas utilizando a assinatura secreta, por isso, prossiga diretamente com a etapa de simulação de recebimento de notificações. Para integrações com Código QR que necessitam verificar a origem das notificações, entre em contato com [Soporte do Mercado Pago](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/support/center).

Author: ```suggestion
> As notificações de Código QR não podem ser verificadas utilizando a assinatura secreta, por isso, prossiga diretamente com a etapa de simulação de recebimento de notificações. Para integrações com Código QR que necessitam verificar a origem das notificações, entre em contato com [Soporte do Mercado Pago](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/support/center).

------------
```

# Webhooks

> Webhooks (também conhecido como retorno de chamada web) são um método simples que permite a um aplicativo ou sistema fornecer informações em tempo real sempre que um evento ocorre. É uma forma passiva de receber dados entre dois sistemas por meio de uma solicitação `HTTP POST`.

Reviewer 2: ```suggestion
Webhooks (também conhecido como retorno de chamada web) são um método simples que permite a uma aplicação ou sistema fornecer informações em tempo real sempre que um evento ocorre. É uma forma passiva de receber dados entre dois sistemas por meio de uma solicitação `HTTP POST`.
```

As notificações Webhooks podem ser configuradas para cada uma das aplicações criadas em  [Suas integrações](/developers/panel/app). Você também poderá configurar uma URL de teste que, junto com suas credenciais de teste, permitirá testar o funcionamento correto das suas notificações antes de sair à produção.

Uma vez configuradas, as notificações Webhooks serão enviadas sempre que ocorrer um ou mais eventos cadastrados. Isso evita a necessidade de verificações constantes, prevenindo a sobrecarga do sistema e a perda de dados em situações críticas. 

Para configurar as notificações Webhooks, escolha entre uma das opções abaixo:

| Tipo de configuração | Descrição |
|---|---|
| [Configuração via Suas integrações](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoviasuasintegraes) | Permite configurar notificações para diversos tópicos, identificar contas diferentes se necessário, e validar a origem da notificação utilizando uma assinatura secreta (exceto para notificações de integrações com Código QR). |

> | [Configuração durante a criação de pagamentos](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoduranteacriaodepagamentos) | Permite a configuração específica das notificações para cada pagamento. Não é permitido configurar para integrações com Mercado Pago Point nem para integrações com Delivery. |

Reviewer 2: ```suggestion
| [Configuração durante a criação de pagamentos](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoduranteacriaodepagamentos) | Permite a configuração específica das notificações para cada pagamento. Não é permitido configurar para integrações com Mercado Pago Point e nem para integrações com Mercado Pago Delivery. |
```

Reviewer 2: Não seria o caso de localizar essa parte que citamos Point e Delivery para os países suportados?

> WARNING
>
> Importante
>
> As URLs configuradas durante a criação do pagamento terão prioridade sobre aquelas configuradas através de Suas integrações. 

Uma vez que as notificações estiverem configuradas, consulte as [ações necessárias após receber uma notificação](/developers/pt/docs/your-integrations/notifications/webhooks#aesnecessriasapsreceberumanotificao) para validar se foram devidamente recebidas.

> 

Reviewer 2: ```suggestion
```

## Configuração via Suas integrações 

> Configure notificações para cada aplicação diretamente em [Suas integrações](/developers/panel/app) de forma eficiente e segura. Nesta documentação, explicaremos como:

Reviewer 2: ```suggestion
Configure notificações para cada aplicação diretamente em [Suas integrações](/developers/panel/app) de forma eficiente e segura. Nesta documentação, explicaremos como:

```

1. Indicar URLs e configurar eventos
2. Validar origem da notificação
3. Simular o recebimento da notificação

> WARNING
>
> Importante
>

> > Este método de configuração não está disponível para integrações com Código QR nem Assinaturas. Para configurar notificações com alguma dessas duas integrações, utilize o método [Configuração durante a criação de pagamentos](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoduranteacriaodepagamentos).

Reviewer 2: ```suggestion
> Este método de configuração não está disponível para integrações com Código QR e nem Assinaturas. Para configurar notificações com alguma dessas duas integrações, utilize o método [Configuração durante a criação de pagamentos](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoduranteacriaodepagamentos).
```

Reviewer 2: Nesse caso, não seria bom localizarmos QR para os países suportados?

    * **URL modo teste:** fornece uma URL que permite testar o correto funcionamento das notificações dessa aplicação durante a fase de teste ou desenvolvimento. O teste dessas notificações deverá ser realizado exclusivamente com as **credenciais de teste de usuários produtivos**.
    * **URL modo produção:** fornece uma URL para receber notificações com sua integração produtiva. Essas notificações deverão ser configuradas com **credenciais produtivas**.

![webhooks](/images/dashboard/webhooks-pt.png)

> NOTE
>
> Nota
> 
> Caso seja necessário identificar múltiplas contas, adicione o parâmetro `?cliente=(nomedovendedor) endpoint` ao final da URL indicada para identificar os vendedores.

3. Selecione os  **eventos** para os quais deseja receber notificações em formato `json` através de um `HTTP POST` para a URL especificada anteriormente. Um evento pode ser qualquer atualização no objeto relatado, incluindo alterações de status ou atributos. Consulte a tabela abaixo para ver os eventos configuráveis, considerando a solução do Mercado Pago integrada e suas necessidades de negócio.

> 

Reviewer 2: ```suggestion
```

| Abertura de _chargebacks_, mudanças de status e modificações referentes às liberações de dinheiro | Chargebacks | `topic_chargebacks_wh` | Checkout Pro<br>Checkout ----[mlb]----Transparente ----------------[mla, mlu, mlc, mlm, mco, mpe]----API------------<br>Checkout Bricks |
----[mla]----| Criação, atualização ou cancelamento de pedidos. | Delivery (proximity marketplace) | `delivery` | MP Delivery |------------

> WARNING
>
> Importante
>
> Em caso de dúvidas sobre quais tópicos ativar ou quais eventos serão notificados, , consulte a documentação [Informações adicionais sobre notificações](/developers/pt/docs/your-integrations/notifications/additional-info). 

5. Por fim, clique e **Salvar** para gerar uma **assinatura secreta** exclusiva para a sua aplicação, permitindo validar a autenticidade das notificações recebidas e garantir que tenham sido enviadas pelo Mercado Pago. A assinatura gerada não tem prazo de validade e sua renovação periódica não é obrigatória, embora seja altamente recomendável. Para renová-la, clique no botão de **redefinição** ao lado da assinatura.

> WARNING
> 
> Importante
> 

> > As notificações de Código QR não podem ser verificadas utilizando a assinatura secreta, por isso, prossiga diretamente com a etapa de simulação de recebimento de notificações. Para integrações com Código QR que necessitam verificar a origem das notificações, entre em contato com [Soporte do Mercado Pago](https://www.mercadopago[FAKER][URL][DOMAIN]/developers/pt/support/center).

Reviewer 2: Não seria o caso de localizar essa nota para os países em que temos QR?

> 

Reviewer 2: ```suggestion
```

### 2. Validar origem da notificação

As notificações enviadas pelo Mercado Pago serão semelhantes ao exemplo abaixo para um alerta do tópico `payment`:

```json
{
 "id": 12345,
 "live_mode": true,
 "type": "payment",
 "date_created": "2015-03-25T10:04:58.396-04:00",
 "user_id": 44444,
 "api_version": "v1",
 "action": "payment.created",
 "data": {
     "id": "999999999"
 }
}
```

O Mercado Pago sempre incluirá a assinatura secreta nas notificações Webhooks recebidas na URL cadastrada. Isso permitirá validar a sua autenticidade, proporcionando maior segurança e prevenindo possíveis fraudes. 

Esta assinatura será enviada no _header_ `x-signature`, conforme o exemplo abaixo.

```x-signature

`ts=1704908010,v1=618c85345248dd820d5fd456117c2ab2ef8eda45a0282ff693eac24131a5e839`

```

Para configurar essa validação, é necessário extrair a chave contida no _header_ e compará-la com a chave fornecida para sua aplicação em Suas integrações. Para isso, siga as etapas abaixo. No final, disponibilizamos alguns SDKs com um exemplo de código completo para facilitar o processo:

> 1. Para extrair o _timestamp_ (`ts`) e a assinatura do _header_ `x-signature`, divida o conteúdo do _header_ pelo caractere `,`, o que resultará em uma lista de 2 elementos. O valor para o prefixo  `ts` é o _timestamp_ (em millissegundos) da notificação, e `v1` é a assinatura encriptada. Seguindo o exemplo apresentado acima, `ts=1704908010` e `v1=618c85345248dd820d5fd456117c2ab2ef8eda45a0282ff693eac24131a5e839`.

Reviewer 2: ```suggestion
1. Para extrair o _timestamp_ (`ts`) e a assinatura do _header_ `x-signature`, divida o conteúdo do _header_ pelo caractere `,`, o que resultará em uma lista de 2 elementos. O valor para o prefixo  `ts` é o _timestamp_ (em millissegundos) da notificação, e `v1` é a assinatura encriptada. Seguindo o exemplo apresentado acima, `ts=1704908010` e `v1=618c85345248dd820d5fd456117c2ab2ef8eda45a0282ff693eac24131a5e839`.
```
```suggestion
1. Para extrair o _timestamp_ (`ts`) e a assinatura do _header_ `x-signature`, divida o conteúdo do _header_ pelo caractere `,`, o que resultará em uma lista de 2 elementos. O valor para o prefixo  `ts` é o _timestamp_ (em milissegundos) da notificação, e `v1` é a assinatura encriptada. Seguindo o exemplo apresentado acima, `ts=1704908010` e `v1=618c85345248dd820d5fd456117c2ab2ef8eda45a0282ff693eac24131a5e839`.
```

```
]]]

### 3. Simular o recebimento da notificação

Para garantir que as notificações estejam configuradas corretamente, é necessário simular o recebimento delas. Para isso, siga os seguintes passos:

1. Após configurar as URLs e os eventos desejados, clique em **Salvar** para salvar a configuração.
2. Após isso, clique em **Simular** para testar se a URL indicada está recebendo as notificações corretamente.
3. Na tela de simulação, selecione a URL a ser testada, podendo ser uma URL de **teste** ou de **produção**.
4. Em seguida, selecione o **tipo de evento** desejado e insira a **identificação** que será enviada no corpo da notificação.
5. Por fim, clique em **Enviar teste** para verificar a solicitação, a resposta dada pelo servidor e a descrição do evento.

## Configuração durante a criação de pagamentos

> Durante o processo de criação de pagamentos, é possível configurar a URL de notificação de maneira específica para cada pagamento, utilizando o campo `notification_url` e implementando o receptor de notificações necessário. 

Author: ```suggestion
Durante o processo de criação de pagamentos, preferências ou ordens comerciais, é possível configurar a URL de notificação de maneira específica para cada pagamento, utilizando o campo `notification_url` e implementando o receptor de notificações necessário. 
```

> 

Author: ```suggestion
----[mla]----
```

> WARNING
>
> Importante
> 
> Este método não permite configurar notificações para os tópicos `point_integration_wh` e `delivery` utilizando este método. Para ativá-los, utilize as [configurações de Suas integrações](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoviasuasintegraes).

> 

Author: ```suggestion

------------

----[mlb, mlm]----

> WARNING
>
> Importante
> 
> Este método não permite configurar notificações para o tópico de `point_integration_wh` utilizando este método. Para ativá-lo, utilize as [configurações de Suas integrações](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoviasuasintegraes).

------------

```

A seguir, explicamos como realizar esta configuração utilizando nossos SDKs.

> 1. No campo `notification_url`, informe a URL que receberá as notificações, conforme o exemplo abaixo. Para receber notificações exclusivamente via Webhooks e não via IPN, adicione o parâmetro `source_news=webhooks` à `notification_url`. Por exemplo: "https://www.yourserver.com/notifications?source_news=webhooks".

Reviewer: ```suggestion
1. No campo `notification_url`, informe a URL que receberá as notificações, conforme o exemplo abaixo. Para receber notificações exclusivamente via Webhooks e não via IPN, adicione o parâmetro `source_news=webhooks` à `notification_url`. Por exemplo: `https://www.yourserver.com/notifications?source_news=webhooks`.
```

```
]]]

### 3. Simular o recebimento da notificação

Para garantir que as notificações estejam configuradas corretamente, é necessário simular o recebimento delas. Para isso, siga os seguintes passos:

1. Após configurar as URLs e os eventos desejados, clique em **Salvar** para salvar a configuração.
2. Após isso, clique em **Simular** para testar se a URL indicada está recebendo as notificações corretamente.
3. Na tela de simulação, selecione a URL a ser testada, podendo ser uma URL de **teste** ou de **produção**.
4. Em seguida, selecione o **tipo de evento** desejado e insira a **identificação** que será enviada no corpo da notificação.
5. Por fim, clique em **Enviar teste** para verificar a solicitação, a resposta dada pelo servidor e a descrição do evento.

## Configuração durante a criação de pagamentos

> Durante o processo de criação de pagamentos, é possível configurar a URL de notificação de  de maneira específica para cada pagamento, utilizando o campo `notification_url` e implementando o receptor de notificações necessário. 

Reviewer 2: ```suggestion
Durante o processo de criação de pagamentos, é possível configurar a URL de notificação de maneira específica para cada pagamento, utilizando o campo `notification_url` e implementando o receptor de notificações necessário. 
```

> WARNING
>
> Importante
> 

> > Este método não permite configurar notificações para os tópicos `point_integration_wh` e `delivery` utilizando este método. Para ativá-los, utilize as [configurações de Suas integrações](/developers/pt/docs/your-integrations/notifications/webhooks#configuraoviasuasintegraes).

Reviewer 2: Não seria o caso de também localizar essa nota?

         $plan = MercadoPago\Invoice::find_by_id($_POST["data"]["id"]);
         break;
     case "point_integration_wh":

>          // $_POST contiene la informaciòn relacionada a la notificaciòn.

Reviewer 2: ```suggestion
         // $_POST contém as informações relacionadas à notificação.
```

         break;
 }
?>
```

> Após realizar as configurações necessárias, a notificação Webhooks será entregue com  formato `JSON`. Veja o exemplo de notificação do tópico de `payments` e as descrições das informações enviadas na tabela abaixo.

Reviewer 2: Em alguns lugares estamos utilizando o termo `JSON` e em outros `json`. É suuuuper um pelo em ovo, mas talvez seja legal padronizarmos. hehe

 }
}
```

| Atributo | Descrição | Exemplo no JSON |
| --- | --- | --- |
| **id** | ID de notificação | `12345` |
| **live_mode** | Indica se a URL informada é válida| `true` |
| **type** | Tipo de notificação recebida de acordo com o tópico previamente selecionado (payments, mp-connect, subscription, claim, automatic-payments, etc.) | `payment` |
| **date_created** | Data de criação do recurso notificado | `2015-03-25T10:04:58.396-04:00` |
| **user_id**| Identificador do vendedor | `44444` |
| **api_version** | Valor que indica a versão da API que envia a notificação. | `v1` |
| **action** | Evento notificado, indicando se é uma atualização de recurso ou a criação de um novo | `payment.created` |
| **data.id**  | ID do pagamento, do `merchant_order` ou da reclamação | `999999999` |

> > WARNING

Author: ```suggestion

----[mla]----
> WARNING
```

>
> Importante
>

> > Para obter o formato de notificações para tópicos diferentes de `payment`, como `point_integration_wh`, `delivery`, `topic_claims_integration_wh` e `topic_card_id_wh`, consulte [Informações adicionais sobre notificações](/developers/pt/docs/your-integrations/notifications/additional-info).

Reviewer: ```suggestion
> Para obter o formato de notificações para tópicos diferentes de `payment`, como `point_integration_wh`, `delivery`, `topic_claims_integration_wh` e `topic_card_id_wh`, consulte [Informações adicionais sobre notificações](/developers/pt/docs/your-integrations/notifications/additional-info).
------------
----[mlb, mlm]----
> WARNING
>
> Importante
>
> Para obter o formato de notificações para tópicos diferentes de `payment`, como `point_integration_wh`, `topic_claims_integration_wh` e `topic_card_id_wh`, consulte [Informações adicionais sobre notificações](/developers/pt/docs/your-integrations/notifications/additional-info).

------------
----[mlu, mlc, mco, mpe]----
> WARNING
>
> Importante
>
> Para obter o formato de notificações para tópicos diferentes de `payment`, como `topic_claims_integration_wh` e `topic_card_id_wh`, consulte [Informações adicionais sobre notificações](/developers/pt/docs/your-integrations/notifications/additional-info).

------------

```

> 

Author: ------------
----[mlb, mlm]----
> WARNING
>
> Importante
>
> Para obter o formato de notificações para tópicos diferentes de `payment`, como `point_integration_wh`, `topic_claims_integration_wh` e `topic_card_id_wh`, consulte [Informações adicionais sobre notificações](/developers/pt/docs/your-integrations/notifications/additional-info).

------------
----[mlu, mlc, mco, mpe]----
> WARNING
>
> Importante
>
> Para obter o formato de notificações para tópicos diferentes de `payment`, como `topic_claims_integration_wh` e `topic_card_id_wh`, consulte [Informações adicionais sobre notificações](/developers/pt/docs/your-integrations/notifications/additional-info).

------------

 }
}
```

| Atributo | Descrição | Exemplo no JSON |
| --- | --- | --- |
| **id** | ID de notificação | `12345` |
| **live_mode** | Indica se a URL informada é válida| `true` |
| **type** | Tipo de notificação recebida de acordo com o tópico previamente selecionado (payments, mp-connect, subscription, claim, automatic-payments, etc.) | `payment` |
| **date_created** | Data de criação do recurso notificado | `2015-03-25T10:04:58.396-04:00` |
| **user_id**| Identificador do vendedor | `44444` |
| **api_version** | Valor que indica a versão da API que envia a notificação. | `v1` |
| **action** | Evento notificado, indicando se é uma atualização de recurso ou a criação de um novo | `payment.created` |
| **data.id**  | ID do pagamento, do `merchant_order` ou da reclamação | `999999999` |

> WARNING
>
> Importante
>

> > Para obter o formato de notificações para tópicos diferentes de `payment`, como `point_integration_wh`, `delivery`, `topic_claims_integration_wh` e `topic_card_id_wh`, consulte [Informações adicionais sobre notificações](/developers/pt/docs/your-integrations/notifications/additional-info).

Reviewer 2: Talvez seja o caso de termos essa nota localizada por país em que o tópico está disponível?

> 

Reviewer 2: ```suggestion
```

------------

Com essas informações, você poderá realizar as atualizações necessárias na sua plataforma como, por exemplo, atualizar um pagamento aprovado.

## Painel de notificações

O painel de notificações permite ao usuário visualizar os eventos disparados sobre uma determinada integração, verificar o status e obter informações detalhadas desses eventos.

Este painel será exibido assim que você configurar suas notificações **Webhooks**, e você pode acessá-lo a qualquer momento clicando em Webhooks dentro de Suas integrações.

Entre as informações disponíveis estão a porcentagem de notificações entregues, bem como uma visão rápida das URLs e dos eventos configurados. 

Além disso, você encontrará uma lista completa das últimas notificações enviadas e seus detalhes, como **status da entrega** (sucesso ou falha), **ação** (ação associada ao evento disparado), **evento** (tipo de evento disparado) e **data e hora**. Se desejar, é possível filtrar esses resultados exibidos por **status da entrega** e por período (**data e hora**).

![paiel de notificações webhooks](/images/dashboard/notification-dashboard-pt.png)

### Detalhes do evento

> Ao clicar em uma das notificações listadas, é possível acessar os detalhes do evento. Esta seção fornece mais informações e permite a recuperação de dados perdidos em caso de falha na entrega da notificação para manter o sistema sempre atualizado.

Reviewer 2: ```suggestion
Ao clicar em uma das notificações listadas, é possível acessar os detalhes do evento. Esta seção fornece mais informações e permite a recuperação de dados perdidos em caso de falha na entrega da notificação para manter o sistema sempre atualizado.

```
